pkg/kds/global/components.go

```diff
@@ -89,26 +91,39 @@ func createZoneIfAbsent(name string, resManager manager.ResourceManager) error {
 			return err
 		}
 		kdsGlobalLog.Info("creating Zone", "name", name)
-		err := resManager.Create(context.Background(), system.NewZoneResource(), store.CreateByKey(name, model.NoMesh))
+		err := resManager.Create(context.Background(), &system.ZoneResource{Spec: &system_proto.Zone{Enabled: true}}, store.CreateByKey(name, model.NoMesh))
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-// ProvidedFilter filter Resources provided by Remote, specifically excludes Dataplanes and Ingresses from 'clusterID' cluster
-func ProvidedFilter(clusterID string, r model.Resource) bool {
-	if r.GetType() == system.ConfigType && r.GetMeta().GetName() != config_manager.ClusterIdConfigKey {
-		return false
-	}
-	if r.GetType() != mesh.DataplaneType {
-		return true
-	}
-	if !r.(*mesh.DataplaneResource).Spec.IsIngress() {
-		return false
+// ProvidedFilter returns ResourceFilter which filters Resources provided by Global, specifically
+// excludes Dataplanes and Ingresses from 'clusterID' cluster
+func ProvidedFilter(rm manager.ResourceManager) reconcile.ResourceFilter {
+	return func(clusterID string, r model.Resource) bool {
+		if r.GetType() == system.ConfigType && r.GetMeta().GetName() != config_manager.ClusterIdConfigKey {
+			return false
+		}
+		if r.GetType() != mesh.DataplaneType {
+			return true
+		}
+		if !r.(*mesh.DataplaneResource).Spec.IsIngress() {
+			return false
+		}
+		if clusterID == util.ZoneTag(r) {
+			// don't need to sync resource to the zone where resource is originated from
+			return false
+		}
+		zone := system.NewZoneResource()
+		if err := rm.Get(context.Background(), zone, store.GetByKey(util.ZoneTag(r), model.NoMesh)); err != nil {
+			kdsGlobalLog.Error(err, "failed to get zone", "zone", util.ZoneTag(r))
+			// since there is no explicit "enabled: false" sync anyway
+			return true
```

Comment:
Yeah, I think it can be false. I wasn't sure about ordering, what if there is no Zone when we build a Snapshot. But I see now that it's not cached or something and we build Snapshot from the scratch every time, so delay with Zone creation won't be a problem

Reply:
Discussed and decided to go with `return true` to avoid unneeded deletions